src/lib/resources/record.resource.ts

```diff
@@ -54,4 +56,66 @@ export class RecordResource extends BaseResource {
 
     return this.getVoFromResponse<RecordResponse>(response, 'RecordVO');
   }
+
+  public async addStorage(record: RecordVO) {
+    const accountResponse = await this.api.account.getSessionAccount();
+    const account = accountResponse.Results[0].data[0].AccountVO;
+    const billingResponse = await this.api.billing.addStorage(
+      account.accountId,
+      record.size
+    );
+    return this.getVoFromResponse<BillingResponse>(
+      billingResponse,
+      'LedgerNonfinancialVO'
+    );
+  }
+
+  public async getPresignedUrl(fileType: string, record: RecordVO) {
+    const response = await this.api.record.getPresignedUrl(fileType, record);
+    if (!response.isSuccessful) {
+      throw new PermSdkError(
+        'could not get presigned url',
+        response.Results[0].message
+      );
+    }
+    return this.getVoFromResponse<SimpleVOResponse>(response, 'SimpleVO');
+  }
+
+  public async registerRecord(record: RecordVO, s3url: string) {
```

Comment:
For both of these methods, the type definition for `record` needs to exclude the `recordId`, since that doesn't exist yet. (I suspect you didn't have a problem while testing because the example code is in JavaScript rather than TypeScript, and this issue comes up during compilation.)